### PR TITLE
remove brfs and use Buffer.from() instead of fs.readFileSync()

### PR DESCRIPTION
**Pull request by ralphtheninja**
Closes https://github.com/Level/levelup/issues/578

### CODE REVIEW

**Comment by vweevers** [comment]
How or where is the fixture used? Maybe it should contain some non-utf8 bytes, to test that the buffer isn't inadvertently stringified. E.g. `Buffer.from('0080c0ff', 'hex')`.

**Comment by ralphtheninja** [comment]
> How or where is the fixture used?

It's used in `test/binary-test.js` via `common.loadBinaryTestData()`

https://github.com/Level/levelup/blob/834c72aa77b1a5efd064f2022bb1244f27d49e52/test/binary-test.js#L13-L21

**Comment by vweevers** [comment]
OK, I see it's used to test both values and keys, so IMO it's important to have non-utf8 bytes.

**Comment by vweevers** [comment]
See [this memdown test](https://github.com/Level/memdown/blob/ee4602b570dc2461f476f3d19b9090d062f5d3a5/test.js#L236) for why it's important (I'm copying that test to `level-js` as well, side note)

**Comment by ralphtheninja** [comment]
Also, it's a bit odd that this code is part of `test/common.js` when it's _only_ used by `test/binary-test.js`

**Comment by ralphtheninja** [comment]
I'll finish this off with some refactoring.

File: test/binary-test.js [hunk 1]
```diff
@@ -150,11 +156,26 @@ buster.testCase('Binary API', {
...
     }.bind(this))
   }
 })
+
+function loadBinaryTestData () {
+  return Buffer.from('0080c0ff', 'hex')
+}
+
+function checkBinaryTestData (testData) {
+  assert.equals(md5sum(testData), binaryTestDataMD5Sum)
```
Comment (ralphtheninja) [inline test/binary-test.js]: We could get rid of `binaryTestDataMD5Sum` as well.

Comment (ralphtheninja) [reply test/binary-test.js]: These functions could now also be renamed to `loadData()` and `checkData()`, since binary is implied within this file.

Comment (ralphtheninja) [reply test/binary-test.js]: > You mean by using buf.equals instead?

I meant rather doing:

```js
function checkData (buf) {
  assert.equals(md5sum(buf), md5sum(loadData()))
}
```

Comment (ralphtheninja) [reply test/binary-test.js]: It would mean we don't cache the md5 sum of the test data, so a bit slower, but cleaner.

**Comment by ralphtheninja** [comment]
Woops, we had an airtap failure I haven't seen before:

```
- starting: <chrome 66 on Windows 10>
events.js:167
      throw er; // Unhandled 'error' event
      ^
Error: ENOSPC: no space left on device, watch '/home/travis/build/Level/[secure]/node_modules/airtap/package.json'
    at FSWatcher.start (fs.js:1407:26)
    at Object.fs.watch (fs.js:1444:11)
    at createFsWatchInstance (/home/travis/build/Level/[secure]/node_modules/chokidar/lib/nodefs-handler.js:37:15)
    at setFsWatchListener (/home/travis/build/Level/[secure]/node_modules/chokidar/lib/nodefs-handler.js:80:15)
    at FSWatcher.NodeFsHandler._watchWithNodeFs (/home/travis/build/Level/[secure]/node_modules/chokidar/lib/nodefs-handler.js:228:14)
    at FSWatcher.NodeFsHandler._handleFile (/home/travis/build/Level/[secure]/node_modules/chokidar/lib/nodefs-handler.js:255:21)
    at FSWatcher.<anonymous> (/home/travis/build/Level/[secure]/node_modules/chokidar/lib/nodefs-handler.js:473:21)
    at FSReqWrap.oncomplete (fs.js:183:5)
Emitted 'error' event at:
    at FSWatcher.emit (events.js:182:13)
    at FSWatcher._handleError (/home/travis/build/Level/[secure]/node_modules/chokidar/index.js:257:10)
    at createFsWatchInstance (/home/travis/build/Level/[secure]/node_modules/chokidar/lib/nodefs-handler.js:39:5)
    at setFsWatchListener (/home/travis/build/Level/[secure]/node_modules/chokidar/lib/nodefs-handler.js:80:15)
    [... lines matching original stack trace ...]
    at FSReqWrap.oncomplete (fs.js:183:5)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! [secure]@3.0.0 test-browsers: `airtap --sauce-connect --loopback airtap.local --no-coverage test/index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the [secure]@3.0.0 test-browsers script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
```

File: test/binary-test.js (hunk 1 above)
Comment (vweevers) [reply test/binary-test.js]: Why do we need to compute a checksum if we can just compare the buffers?

**Comment by vweevers** [comment]
> Woops, we had an airtap failure I haven't seen before

https://github.com/airtap/airtap/issues/152#issuecomment-388552982

**Comment by ralphtheninja** [comment]
Oh right, it's _that_ error, but we haven't released it yet. Lets do that today!

File: test/binary-test.js (hunk 1 above)
Comment (ralphtheninja) [reply test/binary-test.js]: > Why do we need to compute a checksum if we can just compare the buffers?

That's even better. Didn't know about `buf.equals()`!

**Comment by ralphtheninja** [comment]
LOL. This turned into a mega PR :smile:

**Comment by vweevers** [comment]
Ready for review?

**Comment by ralphtheninja** [comment]
> Ready for review?

Yep!

**Comment by ralphtheninja** [comment]
@vweevers You need a new release for `level.js` I take it?

**Comment by vweevers** [comment]
No rush for `level.js`, I fixed it there by adding `brfs` to dependencies.

But it probably does break other people's stuff. `levelup` v3 can't be used in a browser atm.

**Comment by ralphtheninja** [comment]
I'll sneak in an updated airtap and then release a patch.

**Pull request merged**